.git_hooks/pre-commit

```diff
@@ -0,0 +1,166 @@
+#!/bin/bash
+
+# Based on pre-commit taken from https://gist.github.com/wangkuiyi/7379a242f0d4089eaa75
```

Comment:
Is the License compatible with LGPL v2?

Reply:
Added disclaimer of pre-commit hook license in the script

Reply:
It seems that the pre-commit hook is provided under a BSD-like license. I think that is a bit incompatible with the rest of the source, since it restricts that any generated binaries should follow the BSD-like license. This is not possible since the rest of the code should follow LGPL-v2.1. At this point, I would prefer to exclude this file in order to be safe.